### PR TITLE
chore: improve aad-pod-identity errors

### LINKED ISSUE #3610: many "Client.Timeout Exceeded while waiting header" on keda operator log
### Report

when Keda pod launches it starts normally and manage to reconcile all queues and all definition correctly.
there are lot of "healthy" info report .

then after few mintues the log is floaded with "Client.Timeout exceeeded while waiting header".

### Expected Behavior

- we want to understand what this error represent
-  we want to see how and where those timeout can be configured
-  we want to understand why a "healthy" reconcile object turned into timeout error

### Actual Behavior

- those errors do not seem to effect the scale up mechanism , as it seems to continue working 
-  we dont want to see so many error logs without the ability to understand what is the issue and how to fine-tune the timeouts

### Steps to Reproduce the Problem

1. nothing special we just deployed keda on a relative large cluster with many namespaces ,and each namespace has 10-13 queues
2. it could be the developer has created TriggerAutheticationObject Per Queue instead per namespace which caused around 130 TriggerAuthentication object to create -> could that be the problem here ? 
3. Cluster is configured with Managed Identity Support (aad-pod-identity) not with Workload Identity
4. issue happens both on keda 2.7.1 and 2.8.0

### Logs from KEDA operator

```
2022-08-29T19:28:24Z	INFO	Reconciling ScaledObject	{"controller": "scaledobject", "controllerGroup": "keda.sh", "controllerKind": "ScaledObject", "scaledObject": {"name":"fc","namespace":"vi-be-map-dev6"}, "namespace": "vi-be-map-dev6", "name": "fc", "reconcileID": "e1df9a0e-a6bd-4558-bc4e-c07c94aed5d0"}
2022-08-29T19:28:24Z	INFO	Reconciling ScaledObject	{"controller": "scaledobject", "controllerGroup": "keda.sh", "controllerKind": "ScaledObject", "scaledObject": {"name":"tm","namespace":"vi-be-map-dev12"}, "namespace": "vi-be-map-dev12", "name": "tm", "reconcileID": "184cd058-cee7-4975-840b-3f58b8903305"}
2022-08-29T19:28:24Z	INFO	Reconciling ScaledObject	{"controller": "scaledobject", "controllerGroup": "keda.sh", "controllerKind": "ScaledObject", "scaledObject": {"name":"cb","namespace":"vi-be-map-dev6"}, "namespace": "vi-be-map-dev6", "name": "cb", "reconcileID": "ce4c60c5-2772-4be6-a902-785b50af41fa"}
2022-08-29T19:28:25Z	INFO	Reconciling ScaledObject	{"controller": "scaledobject", "controllerGroup": "keda.sh", "controllerKind": "ScaledObject", "scaledObject": {"name":"dp","namespace":"vi-be-map-dev8"}, "namespace": "vi-be-map-dev8", "name": "dp", "reconcileID": "8f3d16c4-6589-44f0-90ee-881eba5aa0d1"}
2022-08-29T19:32:09Z	INFO	Reconciling ScaledObject	{"controller": "scaledobject", "controllerGroup": "keda.sh", "controllerKind": "ScaledObject", "scaledObject": {"name":"aed","namespace":"vi-be-map-dev8"}, "namespace": "vi-be-map-dev8", "name": "aed", "reconcileID": "24b8af3c-c2c4-49c7-ac8b-77fe6123b73c"}
2022-08-29T19:32:09Z	INFO	Initializing Scaling logic according to ScaledObject Specification	{"controller": "scaledobject", "controllerGroup": "keda.sh", "controllerKind": "ScaledObject", "scaledObject": {"name":"aed","namespace":"vi-be-map-dev8"}, "namespace": "vi-be-map-dev8", "name": "aed", "reconcileID": "24b8af3c-c2c4-49c7-ac8b-77fe6123b73c"}
2022-08-29T19:32:10Z	INFO	scaleexecutor	Successfully set ScaleTarget replicas count to ScaledObject minReplicaCount	{"scaledobject.Name": "aed", "scaledObject.Namespace": "vi-be-map-dev8", "scaleTarget.Name": "vi-aed", "Original Replicas Count": 1, "New Replicas Count": 0}
2022-08-29T19:32:21Z	ERROR	azure_servicebus_scaler	error	{"type": "ScaledObject", "namespace": "vi-be-map-dev6", "name": "rc", "error": "Get \"http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&resource=https%3A%2F%2Fservicebus.azure.net%2F\": context deadline exceeded (Client.Timeout exceeded while awaiting headers)"}
github.com/kedacore/keda/v2/pkg/scaling/cache.(*ScalersCache).IsScaledObjectActive
	/workspace/pkg/scaling/cache/scalers_cache.go:89
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).checkScalers
	/workspace/pkg/scaling/scale_handler.go:278
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).startScaleLoop
	/workspace/pkg/scaling/scale_handler.go:149
2022-08-29T19:32:23Z	INFO	Reconciling ScaledObject	{"controller": "scaledobject", "controllerGroup": "keda.sh", "controllerKind": "ScaledObject", "scaledObject": {"name":"aed","namespace":"vi-be-map-dev8"}, "namespace": "vi-be-map-dev8", "name": "aed", "reconcileID": "25e7b3df-cf95-42ef-b170-05fe46187017"}
2022-08-29T19:32:28Z	ERROR	azure_servicebus_scaler	error	{"type": "ScaledObject", "namespace": "vi-be-map-dev12", "name": "frameextraction", "error": "Get \"http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&resource=https%3A%2F%2Fservicebus.azure.net%2F\": context deadline exceeded (Client.Timeout exceeded while awaiting headers)"}
github.com/kedacore/keda/v2/pkg/scaling/cache.(*ScalersCache).IsScaledObjectActive
	/workspace/pkg/scaling/cache/scalers_cache.go:89
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).checkScalers
	/workspace/pkg/scaling/scale_handler.go:278
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).startScaleLoop
	/workspace/pkg/scaling/scale_handler.go:149
2022-08-29T19:32:29Z	ERROR	azure_servicebus_scaler	error	{"type": "ScaledObject", "namespace": "vi-be-map-dev9", "name": "frameextraction", "error": "Get \"http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&resource=https%3A%2F%2Fservicebus.azure.net%2F\": context deadline exceeded (Client.Timeout exceeded while awaiting headers)"}
github.com/kedacore/keda/v2/pkg/scaling/cache.(*ScalersCache).IsScaledObjectActive
	/workspace/pkg/scaling/cache/scalers_cache.go:89
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).checkScalers
	/workspace/pkg/scaling/scale_handler.go:278
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).startScaleLoop
	/workspace/pkg/scaling/scale_handler.go:149
```

### KEDA Version

2.7.1

### Kubernetes Version

1.23

### Platform

Microsoft Azure

### Scaler Details

Azure Service Bus

### Anything else?

- happens both on 2.7.1 and 2.8.0 cluster
-  happens only on relative big clusters 
_No response_

### PR DESCRIPTION
**Pull request by JorTurFer**
Signed-off-by: Jorge Turrado <jorge_turrado@hotmail.es>

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)

Fixes #3610

### CODE REVIEW

**Review by tomkerkhove — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -45,6 +45,7 @@ To learn more about active deprecations, we recommend checking [GitHub Discussio
 ### Improvements
 
 - **General:** Add explicit seccompProfile type to securityContext config ([#3561](https://github.com/kedacore/keda/issues/3561))
+- **General:** Improve error messages in aad-pod-identity integration ([#3610](https://github.com/kedacore/keda/issues/3610))
```
Comment (tomkerkhove) [inline CHANGELOG.md:48]:
```suggestion
- **Azure AD Pod Identity Authentication:** Improve error messages to emphasize problems around the integration with aad-pod-identity itself. ([#3610](https://github.com/kedacore/keda/issues/3610))
```

**Review by tomkerkhove — APPROVED** [review]

**Comment by tomkerkhove** [comment]
@JorTurFer FYI auto-merge won't work

**Comment by JorTurFer** [comment]
Nice... when I applied your suggestion, GH decided to not sign my commit 🤦 
Fixing

**Comment by tomkerkhove** [comment]
Known (annoying) bug that I have reported...

**Comment by JorTurFer** [comment]
Solved! Thanks for notifying! 🙇

**Pull request merged**